recipe/deactivate-clang++.sh

```diff
@@ -49,8 +49,8 @@ function _tc_activation() {
     for thing in "$@"; do
       case "${thing}" in
         *,*)
-          newval=$(echo "${thing}" | sed "s,^[^\,]*\,\(.*\),\1,")
-          thing=$(echo "${thing}" | sed "s,^\([^\,]*\)\,.*,\1,")
+          newval="${thing#*,}"
+          thing="${thing%%,*}"
```

Comment:
For MacOS we neeed
1. Bash 3 compat
2. Bash 4 compat
3. Zsh compat

For Linux we need
1. POSIX sh compat

I don't think this is POSIX sh compatible.

Reply:
It seems string substitution is POSIX (see references), and that string substitution matches sed in a linux machine (see test script).
The string substitution used works like this:
	•	`${parameter#word}`: Remove the smallest prefix pattern.
	•	`${parameter%%word}`: Remove the largest suffix pattern.
References:
[Stackoverflow](https://unix.stackexchange.com/a/389659) 
[POSIX Shell Command Language - Parameter Expansion](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02)

Test script (not output is good)
```sh
#!/bin/sh

test_cases='
""                # Empty string
","               # Single comma
",value1"         # Leading comma
"value1,"         # Trailing comma
"value1"          # Single value
"value1,value2"   # Two values
"value1,,value3"  # Consecutive commas
",,,"             # Only commas
"value1,value2,value3" # Multiple values
"value1,value2,value3,value4"
"value1,value2,,value4"
"value1,,value3,value4"
"value1,,,"
",,,"
"value1,value2,value3,value4,value5,value6"
'

# Remove comments and empty lines
test_cases=$(printf '%s\n' "$test_cases" | sed '/^\s*#/d;/^\s*$/d')

# Loop over each test case
echo "$test_cases" | while IFS= read -r line; do
    thing=$(printf '%s' "$line" | sed 's/^"//;s/"$//') # Remove surrounding quotes

    # Original sed method

    newval_sed=$(echo "${thing}" | sed "s,^[^\,]*\,\(.*\),\1,")
    thing_sed=$(echo "${thing}" | sed "s,^\([^\,]*\)\,.*,\1,")

    # Parameter expansion method
    newval_param="${thing#*,}"
    thing_param="${thing%%,*}"

    # Check if outputs match
    if [ "$newval_sed" = "$newval_param" ] || [ "$thing_sed" = "$thing_param" ]; then
         # echo "Match for input: '$thing'"
         # echo "  Output: thing='$thing_param', newval='$newval_param'"
    else
	echo "Mismatch for input: '$thing'"
        echo "  sed method:      thing='$thing_sed', newval='$newval_sed'"
        echo "  param expansion: thing='$thing_param', newval='$newval_param'"
    fi
done

```

I also captured the values of `$thing` for both a osx and a linux installation , files attached, and used the same test program (write instead `test_cases=$(cat things.txt)`), with no mismatches.
[things_linux.txt](https://github.com/user-attachments/files/17804156/things_linux.txt)
[things_osx.txt](https://github.com/user-attachments/files/17804158/things_osx.txt)